### ISSUE TITLE
Expressions.nullExpresion() in querydsl result in NPE in SqmExpressible

### ISSUE THREAD

**Issue by OHsechun** [spring-projects/spring-data-jpa#3589]
```
kotlin
Spring boot 3.3.2
Spring data jpa 3.3.2 (hibernate 6.5.2)
QueryDSL 5.1.0

java.lang.NullPointerException: Cannot invoke "org.hibernate.type.descriptor.java.JavaType.isInstance(Object)" because the return value of "org.hibernate.query.sqm.SqmExpressible.getExpressibleJavaType()" is null
```

I used Expressions.nullExpression() with case builder in select clause as below.
```
val myCase = QMyEntity.myEntity.field1.`when`("count").then(1).otherwise(Expressions.nullExpression())
(field1 field type is String)

return from(QMyEntity.myEntity)
           .select(myCase.count()`as`("alias") ....
```

In Spring boot 2.7.14 with querydsl 5.0.0 it works well with below query.
```
select count(case when myentity.field1=? then ? else null end) as col_1_0_, ....
from ...
```

How can I fix this issue in Spring data jpa 3.3.2 ?

**Comment by mp911de** [comment]
If you would like us to spend some time helping you to diagnose the problem, please spend some time describing it and, ideally, providing a [minimal yet complete sample](https://stackoverflow.com/help/mcve) that reproduces the problem.
You can share it with us by pushing it to a separate repository on GitHub or by zipping it up and attaching it to this issue.

**Comment by OHsechun** [comment]
@mp911de 
I attached sample project with simple test code.
You can reproduce the problem by running test code.
```
java 21
kotlin 1.9
querydsl 5.1.0
spring boot 3.3.3
```
[demo.zip](https://github.com/user-attachments/files/16755289/demo.zip)

**Comment by mp911de** [comment]
I'm able to reporoduce the same exception with the following JPA code:

```kotlin
val query = entityManager.createQuery(
    "select count(case when sampleEntity.name = ?1 then ?2 else null end)\n" +
            "from SampleEntity sampleEntity"
);

query.setParameter(1, 1)
query.setParameter(2, 1)
```

causing:

```
Cannot invoke "org.hibernate.type.descriptor.java.JavaType.isInstance(Object)" because the return value of "org.hibernate.query.sqm.SqmExpressible.getExpressibleJavaType()" is null
java.lang.NullPointerException: Cannot invoke "org.hibernate.type.descriptor.java.JavaType.isInstance(Object)" because the return value of "org.hibernate.query.sqm.SqmExpressible.getExpressibleJavaType()" is null
	at org.hibernate.query.spi.AbstractCommonQueryContract.isInstance(AbstractCommonQueryContract.java:852)
	at org.hibernate.query.spi.AbstractCommonQueryContract.setParameter(AbstractCommonQueryContract.java:925)
	at org.hibernate.query.spi.AbstractSelectionQuery.setParameter(AbstractSelectionQuery.java:906)
	at org.hibernate.query.sqm.internal.QuerySqmImpl.setParameter(QuerySqmImpl.java:1224)
```

This is a Hibernate-internal issue, please report it against the Hibernate issue tracker https://hibernate.atlassian.net/browse/HHH

**Comment by OHsechun** [comment]
@mp911de 
It was fixed and included at hibernate core 6.6.1, 7.0
Thanks.

https://hibernate.atlassian.net/browse/HHH-18556
